catalog/resource_storage_credential.go

```diff
@@ -76,3 +92,17 @@ func ResourceStorageCredential() *schema.Resource {
 		},
 	}.ToResource()
 }
+
+func waitIAMPropagation(ctx context.Context, c *common.DatabricksClient, f func() error) error {
+	return resource.RetryContext(ctx, IAMPropagationTimeout,
+		func() *resource.RetryError {
+			cerr := f()
+			if e, ok := cerr.(common.APIError); ok && e.StatusCode == 403 && strings.Contains(cerr.Error(), "IAM role") {
```

Comment:
Please make error message matching a bit more concrete

Reply:
I'm now matching on a larger portion of the error message, maybe still not ideal, but hopefully a bit better than before